src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3207,6 +3214,56 @@ private static bool HasNonConstantInitializer(ArrayBuilder<ImmutableArray<FieldO
             AddInitializers(builder.StaticInitializers, staticInitializers);
         }
 
+        private void CollectFieldsFromGlobalDeconstruction(ArrayBuilder<Symbol> builder, VariableComponentSyntax variableComponent,
+            SourceMemberContainerTypeSymbol containingType, VariableComponentAssignmentSyntax assignment)
+        {
+            switch (variableComponent.Kind())
+            {
+                case SyntaxKind.TypedVariableComponent:
+                    CollectFieldsFromGlobalDeconstruction(
```

Comment:
>CollectFieldsFromGlobalDeconstruction [](start = 20, length = 37)

Please collect the type syntax as you descend so you don't have to iterate back up the tree to find it again.